js/app/historyStore.js

```diff
@@ -262,7 +262,7 @@ const HistoryStore = new Lang.Class({
     // the same after a link click, factoring out this common function. When we
     // diverge in future interactions we should revisit this decomposition.
     show_ekn_id: function (ekn_id) {
-        Engine.get_default().get_object_by_id(ekn_id, null, (engine, task) => {
+        Engine.get_default().get_object_by_id(ekn_id, null, null, (engine, task) => {
```

Comment:
Yeah I think with all the C writing I'm getting too used to passing three nulls and a couple 0s.

I'm fine with ditching this last commit entirely, if we can't get it the way we want. Its just our engine API is a bit confusing as is...

Could pass the `app_id` at the end of the arg list here optionally, but it seemed confusing and pattern breaking to put an argument after the cancellable and callback.